compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -772,6 +772,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         let msg = "introduce a new binding instead";
                         let sugg = format!("other_{}", ident.as_str().to_lowercase());
                         e.span_suggestion(ident.span, msg, sugg, Applicability::HasPlaceholders);
+                        match res {
+                            Res::Def(DefKind::Const, _) => {
+                                let const_msg = "Constants currently only support matching the same type. \
+                                This may support for structural matches in the future. \
+                                See https://github.com/rust-lang/rust/issues/76001.";
```

Comment:
https://github.com/rust-lang/rust/issues/76001 does not look like it will ever support this. The RFC does not mention structural matches.

Reply:
would https://github.com/rust-lang/rust/issues/74446 be better?

Reply:
I don't think so. I don't think this is supposed to work.

Reply:
Ok, ill ask in zulip and try to make it so the diagnostic checks if its a range type